python/ray/util/data/__init__.py

```diff
@@ -0,0 +1,115 @@
+from collections import defaultdict
+from typing import List, Iterable
+
+import pandas as pd
+
+from ray.util.data.dataset import MLDataset
+from ray.util.data.parquet import read_parquet
+from ray.util.data.source import SourceShard
+from ray.util.iter import T, ParallelIterator
+import ray.util.iter as parallel_it
+
+try:
+    import dataclasses
+except:
+    pass
+else:
+    from dataclasses import is_dataclass
+
+
+def to_pandas(it: ParallelIterator[T],
+              batch_size: int = 32) -> "ParallelIterator[pd.DataFrame]":
+    """
+    Convert the a ParallelIterator to ParallelIterator of pd.DataFrame. The
+    record type should be list like object or dataclass instance. If
+    the record is a iterable, we will convert to a list. If the record has
+    __getitem__ attr, we will use __getitem__ to get the given column
+    indexes data to create pandas DataFrame. If the record is dataclass
+    instance we will use __getattr__ to get the given column.
+    Args:
+        it (ParallelIterator[T]): the ParallelIterator to converted
+        batch_size (int): batch the given size to create a pandas DataFrame
+    Returns:
+        A ParallelIterator of pd.DataFrame
+    """
+    it = it.batch(batch_size)
+
+    def convert_fn(input_it: Iterable[T]) -> Iterable[pd.DataFrame]:
+        names = []
+        for batch in input_it:
+            assert isinstance(batch, list)
+            if hasattr(batch[0], "__getitem__"):
+                batch = pd.DataFrame(batch)
+            elif hasattr(batch[0], "__iter__"):
+                batch = [list(item) for item in batch]
+                batch = pd.DataFrame(batch)
+            elif is_dataclass(batch[0]):
+                if not names:
+                    names = [f.name for f in dataclasses.fields(batch[0])]
+                values = defaultdict(lambda x: [])
+                for item in batch:
+                    for col in names:
+                        values[col].append(getattr(item, col))
+                batch = pd.DataFrame(values, columns=names)
+            else:
+                raise ValueError(
+                    "DistributedDataset only support list like "
```

Comment:
DistributedDataset needs to be renamed here and in a few other comments as well

Reply:
fixed